pandas/_libs/tslibs/src/datetime/np_datetime_strings.c

```diff
@@ -156,13 +201,21 @@ int parse_iso_8601_datetime(const char *str, int len, int want_exc,
         ymd_sep = valid_ymd_sep[i];
         ++substr;
         --sublen;
+
+        const char tmp[1] = {ymd_sep};
+        if (compare_format(&format, &format_len, tmp, 1, exact)) {
```

Comment:
```suggestion
        if (compare_format(&format, &format_len, &ymd_sep, 1, exact)) {
```

The address of a char gives you a char * that you need. If you do this can delete the preceding line

Reply:
ah that works, thanks!